src/bentoml/_internal/configuration/helpers.py

```diff
@@ -105,17 +105,17 @@ def get_default_config(version: int) -> dict[str, t.Any]:
         )
     )
     mod = import_configuration_spec(version)
-    assert hasattr(mod, "SCHEMA"), (
-        "version %d does not have a validation schema" % version
-    )
     try:
         mod.SCHEMA.validate(config)
     except s.SchemaError as e:
         raise BentoMLConfigException(
             "Default configuration for version %d does not conform to given schema:\n%s"
             % (version, e)
-        ) from None
-    return config
+        ) from e
+    except Exception as err:
+        raise BentoMLException("Unexpected error while setting configuration:\n") from err
+    finally:
+        return config
```

Comment:
Does this finally actually do anything? Isn't it the same as just having return on the inside?

Reply:
Yes, this is incorrect, the config will be always returned and no exception will be raised.